lib/src/repository.rs

```diff
@@ -276,46 +350,104 @@ impl OmicronRepoEditor {
             repo_path,
             artifacts: ArtifactsDocument::empty(system_version),
             existing_target_names: BTreeSet::new(),
+            existing_deployment_units: DeploymentUnitMapBuilder::new(
+                DeploymentUnitScope::Repository,
+            ),
         })
     }
 
     /// Adds an artifact to the repository.
-    pub fn add_artifact(&mut self, new_artifact: &AddArtifact) -> Result<()> {
+    pub fn add_artifact(
+        &mut self,
+        new_artifact: &AddArtifact,
+    ) -> Result<ArtifactHash> {
         let target_name = format!(
             "{}-{}-{}.tar.gz",
             new_artifact.kind(),
             new_artifact.name(),
             new_artifact.version(),
         );
 
-        // make sure we're not overwriting an existing target (either one that
+        let mut errors = Vec::new();
+
+        // Make sure we're not overwriting an existing target (either one that
         // existed when we opened the repo, or one that's been added via this
         // method)
         if !self.existing_target_names.insert(target_name.clone()) {
-            bail!(
+            errors.push(anyhow!(
```

Comment:
With this change will we now overwrite an existing target?

Reply:
Fixed, thanks!